plugins/action/zos_copy.py

```diff
@@ -150,14 +160,59 @@ def run(self, tmp=None, task_vars=None):
                             dict(src=src, dest=dest, changed=False, failed=True)
                         )
                         return result
+
+                    if use_template:
+                        try:
+                            template_parameters = task_args.get("template_parameters", dict())
+                            if encoding:
+                                template_encoding = encoding.get("from", None)
+                            else:
+                                template_encoding = None
+
+                            renderer = template.create_template_environment(
+                                template_parameters,
+                                src,
+                                template_encoding
+                            )
+                            template_dir, rendered_dir = renderer.render_dir_template(
+                                task_vars.get("vars", dict())
+                            )
+                            src = rendered_dir
+                        except Exception as err:
+                            return self._exit_action(result, str(err), failed=True)
+
                     task_args["size"] = sum(
-                        os.stat(path + "/" + f).st_size for f in files
+                        os.stat(os.path.join(path, f)).st_size
+                        for path, dirs, files in os.walk(src)
+                        for f in files
                     )
                 else:
                     if mode == "preserve":
                         task_args["mode"] = "0{0:o}".format(
                             stat.S_IMODE(os.stat(src).st_mode)
                         )
+
+                    if use_template:
+                        try:
```

Comment:
same comment as above.

Reply:
Not needed as "I moved down the try block. It still includes the creation and assigment of the TemplateRenderer object because its __init__ method can throw two exceptions."